crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -200,11 +236,45 @@ impl EncString {
         Ok(EncString::AesCbc256_HmacSha256_B64 { iv, mac, data })
     }
 
+    pub(crate) fn encrypt_xchacha20_poly1305(
+        data_dec: &[u8],
+        key: &XChaCha20Poly1305Key,
+    ) -> Result<EncString> {
+        let mut protected_header = coset::HeaderBuilder::new().build();
+        protected_header.alg = Some(coset::Algorithm::PrivateUse(cose::XCHACHA20_POLY1305));
+
+        let mut nonce = [0u8; 24];
+        let cose_encrypt0 = coset::CoseEncrypt0Builder::new()
+            .protected(protected_header)
+            .try_create_ciphertext(data_dec, &[], |data, aad| {
+                let ciphertext = crate::xchacha20::encrypt_xchacha20_poly1305(
+                    key.enc_key
+                        .as_slice()
+                        .try_into()
+                        .expect("XChaChaPoly1305 key is 32 bytes long"),
+                    data,
+                    aad,
+                );
```

Comment:
I'm preetty sure we can just `into()` a `GenericArray<u8, U32>` into a `[u8; 32]`:
```suggestion
                let ciphertext =
                    crate::xchacha20::encrypt_xchacha20_poly1305(&(*key.enc_key).into(), data, aad);
```

I think theres a couple of other cases of `.as_slice().try_into().expect("")` that could also be changed over.

Reply:
Replaced the instances for the enckey for encrypt and decrypt for now.